logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/type/UserTypeConverter.kt

```diff
@@ -0,0 +1,27 @@
+package com.wire.kalium.logic.data.user.type
+
+import com.wire.kalium.persistence.dao.UserTypeEntity
+
+interface UserTypeConverter<T> {
+    val guest: T
+    val federated: T
+    val internal: T
+}
+
+class DomainUserTypeConverter : UserTypeConverter<UserType> {
+    override val guest: UserType
+        get() = UserType.GUEST
+    override val federated: UserType
+        get() = UserType.FEDERATED
+    override val internal: UserType
+        get() = UserType.INTERNAL
+}
+
+class EntityUserTypeConverter : UserTypeConverter<UserTypeEntity> {
```

Comment:
Why this is done in other way than other entities which aren't children of the same class?

Reply:
my idea was to reuse the logic inside UserTypeMapper, so I created a Converter interface which would delegate the logic to which enum we convert the value, because we have two 
1. for the entity 
2. for the domain object

and each of them depends on: 
   otherUserDomain: String,
        selfUserTeamId: String?,
        otherUserTeamID: String?
        
        and has same logic, just returns a different object

Reply:
✅